src/adaptors/tcp/tcp_adaptor.c

```diff
@@ -2234,6 +2234,7 @@ static void qdr_tcp_activate_CT(void *notused, qdr_connection_t *c)
  */
 static void qdr_tcp_adaptor_init(qdr_core_t *core, void **adaptor_context)
 {
+    qd_adaptor_common_init();
```

Comment:
Looks like we are doing this tiered approach only in the tcp adaptor, just making sure ?